data-avro/src/main/java/com/linkedin/data/avro/PegasusUnionToAvroRecordConvertCallback.java

```diff
@@ -62,6 +63,14 @@ public void callback(List<String> path, DataSchema schema)
     {
       DataSchema fieldSchema = field.getType().getDereferencedDataSchema();
 
+      Map<String, Object>  propagatedProperties = fieldSchema.getProperties();
```

Comment:
I wonder if you can just call `getMergedTyperefProperties`

like 
`Map<String, Object>  propagatedProperties = fieldSchema.getMergedTyperefProperties();`

Because looking into the `getMergedTyperefProperties` function it will do the judge for you if it is a TypeRef

Reply:
Hi Brian,
I think you misunderstood. The branch with judge "(getRef().getType() == Type.TYPEREF)" is for a recursive call.

this `getMergedTyperefProperties` is only available in `TyperefDataSchema`

Reply:
Thanks, You are right, I didnt fint it a typeref specific method